dotnet-engine/Yggdrasil.Engine/Types.cs

```diff
@@ -13,29 +13,31 @@ public class Context
     public Dictionary<string, string>? Properties { get; set; }
 }
 
-public class EngineResponse {
+public class EngineResponse
+{
     [JsonPropertyName("error_message")]
     public string? ErrorMessage { get; set; }
 
     [JsonPropertyName("status_code")]
     public string? StatusCode { get; set; }
 }
 
-public class EngineResponse<TValue> : EngineResponse {
+public class EngineResponse<TValue> : EngineResponse
+{
     public TValue? Value { get; set; }
 }
 
-public class Variant
+public sealed class Variant
 {
-    public string Name { get; set; }
+    public string Name { get; set; } = null!;
```

Comment:
Sneaky .NET syntax - we assign a default value of null but tell the compiler that this will never be the case. I think this is okay here since these are always constructed through deserialisation

Reply:
weird, maybe @daveleek can help with this better than I can

Reply:
The other alternative is to use constructors with non-nullable parameters - which should cause exceptions on deserialization